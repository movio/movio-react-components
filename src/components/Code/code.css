@import "variables.css";

:root {
  --color-bg: var(--color-neutral-2);
  --color-text: var(--color-primary-d);
}

.container {
  background: var(--color-bg);
  color: white;
  padding: 1rem;
  border-radius: var(--border-radius);
}

.code {
  color: var(--color-text);
}
