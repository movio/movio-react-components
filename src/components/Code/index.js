import React, { PropTypes } from 'react';
import classnames from 'classnames';

import styles from './code.css';

const Code = ({ className, code }) =>
  <pre className={classnames(className, styles.container)}>
    <code className={styles.code}>
      {code}
    </code>
  </pre>;

Code.propTypes = {
  className: PropTypes.string,
  code: PropTypes.string.isRequired,
};


export default Code;
export {
  styles,
};
