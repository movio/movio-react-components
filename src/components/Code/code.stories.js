import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text, select } from '@storybook/addon-knobs';
import Code from './index';

import utilStyles from '../../../.storybook/utils.css';

const stories = storiesOf('Code', module);

stories.addDecorator(withKnobs);

const exampleCode = `const add = (a, b) => a + b;
const map = (fn, xs) => xs.map(fn);

.container {
  background: var(--color-bg);
  color: white;
  padding: 1rem;
  border-radius: var(--border-radius);
}

.code {
  color: var(--color-text);
}
`;

stories.add('base', () => <Code code={exampleCode} />).add('with knobs', () => (
  <Code
    code={text('Code', exampleCode)}
    className={select(
      'Custom Clasnames',
      {
        [utilStyles.small]: 'Small',
        [utilStyles.large]: 'Large',
        0: 'None',
      },
      0
    )}
  />
));
