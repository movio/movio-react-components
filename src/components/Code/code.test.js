import React from 'react';
import { expect } from 'chai';
import { shallow } from 'enzyme';

import Code, { styles } from './index';

describe('<Code />', () => {
  it('should render a <code> block as the only child', () => {
    const wrapper = shallow(<Code code="const a = (b) => b + 1;" />);
    expect(wrapper.children()).to.have.length(1);
    expect(wrapper.find('code')).to.have.length(1);
  });

  it('should append a className to the default, if provided', () => {
    const wrapper = shallow(<Code className="test-class" code="const a = (b) => b + 1;" />);
    expect(wrapper.prop('className')).to.equal(`test-class ${styles.container}`);
  });

  it('should use the default className from the stylesheet, if not provided', () => {
    const wrapper = shallow(<Code code="const a = (b) => b + 1;" />);
    expect(wrapper.prop('className')).to.equal(styles.container);
  });

  it('should correctly render code block', () => {
    const wrapper = shallow(<Code code="const a = (b) => b + 1;" />);
    expect(wrapper.find('code').text()).to.equal('const a = (b) => b + 1;');
  });
});

