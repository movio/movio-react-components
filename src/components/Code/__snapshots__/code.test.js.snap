// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`code-custom-classname 1`] = `
<pre
  className="test-class container"
>
  <code
    className="code"
  >
    const a = (b) =&gt; b + 1;
  </code>
</pre>
`;

exports[`code-default-classname 1`] = `
<pre
  className="container"
>
  <code
    className="code"
  >
    const a = (b) =&gt; b + 1;
  </code>
</pre>
`;

exports[`code-single-child 1`] = `
<pre
  className="container"
>
  <code
    className="code"
  >
    const a = (b) =&gt; b + 1;
  </code>
</pre>
`;
